lua/rustaceanvim/dap.lua

```diff
@@ -120,16 +121,85 @@ local function generate_source_map()
   end)
 end
 
----@param args RADebuggableArgs
-function M.start(args)
-  vim.notify('Compiling a debug build for debugging. This might take some time...')
+---@type {[string]: string}
+local init_commands = {}
+
+local function get_lldb_commands(workspace_root)
+  get_rustc_sysroot(function(rustc_sysroot)
+    local script_import = 'command script import "'
+      .. compat.joinpath(rustc_sysroot, 'lib', 'rustlib', 'etc', 'lldb_lookup.py')
+      .. '"'
+    local commands_file = compat.joinpath(rustc_sysroot, 'lib', 'rustlib', 'etc', 'lldb_commands')
+    local file = io.open(commands_file, 'r')
+    local workspace_root_cmds = {}
+    if file then
+      for line in file:lines() do
+        table.insert(workspace_root_cmds, line)
+      end
+      file:close()
+    end
+    table.insert(workspace_root_cmds, 1, script_import)
+    init_commands = vim.tbl_extend('force', init_commands, { [workspace_root] = workspace_root_cmds })
+  end)
+end
+
+---@type string[]
+local environment = {}
 
+-- Most succinct description: https://github.com/bevyengine/bevy/issues/2589#issuecomment-1753413600
+---@param workspace_root string
+local function add_dynamic_library_paths(workspace_root)
+  compat.system({ 'rustc', '--print', 'target-libdir' }, nil, function(sc)
+    ---@cast sc vim.SystemCompleted
+    local result = sc.stdout
+    if sc.code ~= 0 or result == nil then
+      return
+    end
+    local rustc_target_path = (result:gsub('\n$', ''))
+    local target_path = compat.joinpath(workspace_root, 'target', 'debug', 'deps')
+    local sep = ':'
+    local win_sep = ';'
+    if shell.is_windows() then
+      table.insert(environment, 'PATH=' .. rustc_target_path .. win_sep .. target_path .. win_sep .. '%PATH%')
+    elseif shell.is_macos() then
+      table.insert(
+        environment,
+        'DKLD_LIBRARY_PATH=' .. rustc_target_path .. sep .. target_path .. sep .. '$DYLD_LIBRARY_PATH'
+      )
+    else
+      table.insert(
```

Comment:
issue: As with `init_commands` previously, `environment` will grow indefinitely. We can prevent this in the same way, by using a table with the workspace root as the key.

Reply:
Shoot, yep ty will fix.

Reply:
Oh nvm, I see you're on it :) I'll keep looking at the launch.json thing.